ruma-identifiers-validation/src/crypto_algorithms.rs

```diff
@@ -65,9 +66,58 @@ impl TryFrom<String> for ServerKeyAlgorithm {
     }
 }
 
+/// An encryption algorithm to be used to encrypt messages sent to a room.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
+#[serde(from = "String", into = "String")]
```

Comment:
The serde derives and attributes need to be feature-gated, like they are for the other enums here.